src/realm/object-store/c_api/logging.cpp

```diff
@@ -36,15 +36,18 @@ static_assert(realm_log_level_e(Logger::Level::off) == RLM_LOG_LEVEL_OFF);
 class CLogger : public realm::util::Logger {
 public:
     CLogger(UserdataPtr userdata, realm_log_func_t log_callback, Logger::Level level)
-        : Logger(level)
+        : Logger()
         , m_userdata(std::move(userdata))
         , m_log_callback(log_callback)
     {
+        set_level_threshold(level);
     }
 
 protected:
-    void do_log(Logger::Level level, const std::string& message) final
+    void do_log(const util::LogCategory&, Logger::Level level, const std::string& message) final
     {
+
+        // FIXME use category
```

Comment:
so how the capi would change to accommodate this?

Reply:
The logger function must be able to receive a string indicating the category